src/prefs/WaveformSettings.h

```diff
@@ -81,4 +89,34 @@ class AUDACITY_DLL_API WaveformSettings : public PrefsListener
    // Convenience
    bool isLinear() const { return stLinear == scaleType; }
 };
+
+class AUDACITY_DLL_API WaveformSettingsCache
+   : public ClientData::Cloneable< ClientData::UniquePtr >
+{
+public:
+
+   //! Get either the global default settings, or the track's own if previously created
+   static WaveformSettingsCache &Get( WaveTrack &track );
+
+   //! @copydoc Get
+   static const WaveformSettingsCache &Get( const WaveTrack &track );
+
+   ~WaveformSettingsCache() override;
+   PointerType Clone() const override;
+
+   int ZeroLevelYCoordinate(wxRect rect) const;
+
+   void GetDisplayBounds(float &min, float &max) const
+   { min = mDisplayMin; max = mDisplayMax; }
+
+   void SetDisplayBounds(float min, float max) const
+   { mDisplayMin = min; mDisplayMax = max; }
+
+   float GetLastScaleType() const { return mLastScaleType; }
+
+   mutable float mDisplayMin = -1.0f, mDisplayMax = 1.0f;
```

Comment:
Why the fields are public?

Reply:
No good reason here.  I can change it.

Though I do not agree that whenever there is a structure, you must encapsulate everything with get and set functions.  Some things are simply bags of data with no important consistency constraints, no invariants, to be established and preserved by member functions.  Such for instance is the case with many structure names ending in `...Options` that are just meant to shorten long lists of arguments.

Reply:
I have a pretty strong opinion, that mixing POD-like objects and getters/setters is not a best idea. Also in most of the places `m` prefix in Audacity means some kind of encapsulation. It is frustrating to read and understand the code when naming conventions are violated.

Reply:
Arguably this is not a POD because there should be an ordering constraint on mins and maxes.

However I have my equally strong opinion that one should not be slavish about the `m` convention.  Some PODs really are PODs.  A structure with no invariant conditions it needs to protect is just a bunch of stuff and not properly a class.

Reply:
The fact that something does NOT use the `m` conventions should be its own recognized convention too.  That, this structure is meant to be just a POD.

Reply:
> Arguably this is not a POD because there should be an ordering constraint on mins and maxes.

Public access to members removes any constraints.